### ISSUE TITLE
Crash around 8GB

### ISSUE THREAD

**Issue by edencaldas** [theBlackDragon/conan-exiles-dedicated-server#5]
Have you ever passed 8GB of RAM usage with it?

**Comment by theBlackDragon** [comment]
Alas, I've started running into crashing issues after one of the last big patches as well. I can't say I successfully went past 8GiB in the past or not since I wasn't tracking it specifically. Arguably, in my case the fact that it goes past 8GiB is already a problem in itself, as it doesn't on Windows.

I've tried switching explicitly to 64bit wine (which it should already have been using regardless), and explicitly installing Mono into the VM but neither really helped any. I can push those changes to a branch if you want to see if they help any for your case.

Any other suggestions are more than welcome at this point.

**Comment by edencaldas** [comment]
Hi, I've tried everything. I don't think there is anything you can do to change your docker image to fix that.

I've tried
- Centos 7, Ubuntu 20.04. Debian 10
- Wine from distro repos, Wine from wine-hq repos, compiled wined.
- Running in host, running in container.
- In container, I've tried pinning CPUs, limiting memory.
- Messing with Conan Exiles INI files and changing memory parameters (they are not well documented)
- Tried turning off Unreal Engine 4 garbage collaction.

Nothing worked.

I see this same issue being reported in other conan docker releases here in github, as well as in the conan exiles admins discord channel. No one has a fix.

I've also contacted Funcom through the forums, they asked for a few log files of mine which I've provided but haven't heard of them yet.

If I ever find a solution I will inform you.

**Comment by theBlackDragon** [comment]
Thanks would appreciate it if a solution would pop up.

Wish they'd finally just make a native Linux version of the server. Certainly would make more sense than shelling out for Windows, or burning the additional Watts to run it in a Windows VM.

**Comment by DubLDub** [comment]
I assume there are no positive updates on this? Running into a lot more now with purges.

I tested with Wine64 and Bookworm, but doesn't help too much.

**Comment by theBlackDragon** [comment]
Not really, unfortunately, I've been trying to sort out a working container based on an updated Debian image on and off with rather little success :(

**Comment by kodi022** [comment]
I am having these issues too, modlist runs no issue on singleplayer, but on the server (debian, wine64) once i reach around 8 to 8.1GB ram usage, guaranteed lock up after  "Freeing ... bytes from backup pool to handle out memory"

Really hope some solution can be found

**Comment by kodi022** [comment]
Just to give a little update to this, and disclaimer, I do not use docker or this repo for the Conan Exiles dedicated server, but I had that seemingly identical issue just trying to start it through wine. Searching for solutions online is how I stumbled across here. So my solution might not be relevant here, but I thought its best to share it somewhere.

The solution for my issue was to remove Debian's sources version of win `(8.0~repack-4)` and installing the newest version I could `(9.4~bookworm-1)`. Don't know why or how it makes a difference for me, but now I have been able to play on my server that contains 40+ mods including AoC.
You can reference here if your looking to do the same with Debian https://wiki.winehq.org/Debian

I worry it didn't truly solve the issue, but instead perhaps its just more efficient with ram? Possibly postponing the crash, as I have not seen it go above 9.7GB yet, but its been issue free with 2 people for 3+ hours as of writing.

**Comment by DubLDub** [comment]
@kodi022 - Can confirm, made the change to install winehq-staging instead (installs 9.5), and that has completely resolved the 8GB crashing issues. Thanks!
